api/controllers/v0.2/createPatron.js

```diff
@@ -55,74 +55,155 @@ function renderResponse(req, res, status, message) {
+        `message: "${response.message} from ILS.", ` +
+        `response: ${JSON.stringify(response.response.data)}` // eslint-disable-line comma-dangle
+      );
+
+      if (response.response && response.response.data) {
+        const responseObject = collectErrorResponseData(
+          response.response.status,
+          response.response.data.type,
+          response.response.data.detail,
+          response.response.data.title,
+          response.response.data.debug_message // eslint-disable-line comma-dangle
+        );
+
+        const statusCode = (responseObject.status) ? responseObject.status : 500;
+
+        renderResponse(
+          req,
+          res,
+          statusCode,
+          modelResponse.errorResponseData(responseObject) // eslint-disable-line comma-dangle
+        );
+      } else {
+        renderResponse(req, res, 500, modelResponse.errorResponseData(
+          collectErrorResponseData(null, '', '', '', '') // eslint-disable-line comma-dangle
+        ));
+      }
+    });
+}
 
-  if (!process.env.ILS_CREATE_PATRON_URL || isEmpty(process.env.ILS_CREATE_PATRON_URL)) {
+/**
+ * validateEnvVariable(envVariableName)
+ * Validate that a specific environment variable is present.
+ *
+ * @param {HTTP response} res
+ * @param {string} envVariableName
+ */
+function validateEnvVariable(res, envVariableName) {
+  const envVariable = process.env[envVariableName];
+
+  if (!envVariable || isEmpty(envVariable)) {
     res
       .status(500)
       .header('Content-Type', 'application/json')
       .json(modelResponse.errorResponseData(
         collectErrorResponseData(
           null,
           'invalid-request',
-          'The create-patron service is missing the ILS_CREATE_PATRON_URL.',
+          `The create-patron service is missing ${envVariableName}.`,
           null,
           { form: ['The create-patron service is misconfigured.'] } // eslint-disable-line comma-dangle
         ) // eslint-disable-line comma-dangle
       ));
-
-    return;
+    return false;
   }
+  return true;
+}
 
-  const patronData = req.body;
-
-  // eslint-disable-next-line max-len
-  // Delete pcode4 data to prevent an error on Sierra TODO: remove this line once pcode4 is accepted; also remove related line in v0.2 modelStreamPatron.js
-  delete patronData.patronCodes.pcode4;
-
-  const requiredFields = [];
+/**
+ * validateEnvironmentAndRequest(req, res)
+ * Validate the environment variables and the request.
+ *
+ * @param {HTTP request} req
+ * @param {HTTP response} res
+ */
+function validateEnvironmentAndRequest(req, res) {
+  const envVariableNames = [
+    'ILS_CLIENT_KEY', 'ILS_CLIENT_SECRET', 'SCHEMA_API_BASE_URL', 'ILS_CREATE_PATRON_URL',
+    'ILS_CREATE_TOKEN_URL', 'PATRON_STREAM_NAME_V02', 'PATRON_SCHEMA_NAME_V02',
+  ];
+  envVariableNames.forEach((envVariableName) => {
+    if (validateEnvVariable(res, envVariableName) === false) {
+      return false;
+    }
+    return true;
+  });
```

Comment:
`validateEnvVariable` as written validates by writing directly to the HTTPResponse object. I don't think there's any utility checking the return value of `validateEnvVariable` as written. In any case, returning from within a `forEach` callback has no effect here. Is the desire that you write a `400` response for the first missing ENV var and then stop processing?